src/checkers.cpp

```diff
@@ -35,7 +34,7 @@ namespace ada::checkers {
     }
 
     if (parts_count > 1) {
-      pointer_start = std::find_end(pointer_start, pointer_end, delimiter.begin(), delimiter.end());
+      pointer_start = pointer_start + std::string_view{pointer_start, size_t(pointer_end - pointer_start)}.rfind(delimiter);
```

Comment:
Isn't this much more costly than the previous implementation?

Reply:
`pointer_start + std::string_view{pointer_start, size_t(pointer_end - pointer_start)}.rfind(',')` compiles to this...
```asm
        sub     rdx, rdi
        lea     rax, [rdi-1]
        lea     rdx, [rdi-1+rdx]
        jne     .L26
        jmp     .L33
.L29:
        lea     rsi, [rdx-1]
        cmp     cl, BYTE PTR [rdx]
        je      .L28
        mov     rdx, rsi
.L26:
        cmp     rdx, rax
        jne     .L29
        ret
.L33:
        lea     rax, [rdi-1]
        ret
.L28:
        mov     rax, rdx
        ret
```

... You may be able to do better, but it looks like very tight and cheap code.

Meanwhile, ` pointer_start = std::find_end(pointer_start, pointer_end, delimiter.begin(), delimiter.end());` compiles to the following monster...

```asm
        push    r14
        push    r13
        push    r12
        mov     r12, rdx
        push    rbp
        push    rbx
        mov     rbx, rdi
        sub     rsp, 32
        mov     rbp, QWORD PTR [rcx]
        mov     r14, QWORD PTR [rcx+8]
        lea     r13, [rbp+0+r14]
        cmp     rdi, rdx
        je      .L43
        cmp     rbp, r13
        je      .L43
        lea     rax, [r13-1]
        mov     rsi, rdx
        cmp     rbp, rax
        je      .L46
.L36:
        mov     QWORD PTR [rsp+8], rsi
        lea     rcx, [rsp+24]
        lea     rdx, [rsp+16]
        mov     rdi, rsp
        lea     rsi, [rsp+8]
        mov     QWORD PTR [rsp+24], r13
        mov     QWORD PTR [rsp+16], rbx
        call    std::reverse_iterator<char const*> std::__find_if<std::reverse_iterator<char const*>, __gnu_cxx::__ops::_Iter_equals_iter<std::reverse_iterator<__gnu_cxx::__normal_iterator<char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >(std::reverse_iterator<char const*>, std::reverse_iterator<char const*>, __gnu_cxx::__ops::_Iter_equals_iter<std::reverse_iterator<__gnu_cxx::__normal_iterator<char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, std::random_access_iterator_tag) [clone .isra.0]
        mov     rax, QWORD PTR [rsp]
        cmp     rbx, rax
        je      .L39
        lea     rsi, [rax-1]
        cmp     rbx, rsi
        je      .L39
        lea     rdx, [r13-2]
        mov     rcx, rsi
.L40:
        movzx   edi, BYTE PTR [rdx]
        cmp     BYTE PTR [rcx-1], dil
        jne     .L36
        cmp     rbp, rdx
        je      .L41
        sub     rcx, 1
        sub     rdx, 1
        cmp     rbx, rcx
        jne     .L40
.L39:
        add     rsp, 32
        mov     rax, r12
        pop     rbx
        pop     rbp
        pop     r12
        pop     r13
        pop     r14
        ret
.L43:
        mov     rax, r12
.L35:
        cmp     rbx, rax
        je      .L39
.L41:
        add     rsp, 32
        sub     rax, r14
        pop     rbx
        pop     rbp
        pop     r12
        pop     r13
        pop     r14
        ret
.L46:
        mov     QWORD PTR [rsp+16], rdi
        lea     rcx, [rsp+24]
        lea     rsi, [rsp+8]
        mov     rdi, rsp
        mov     QWORD PTR [rsp+8], rdx
        lea     rdx, [rsp+16]
        mov     QWORD PTR [rsp+24], r13
        call    std::reverse_iterator<char const*> std::__find_if<std::reverse_iterator<char const*>, __gnu_cxx::__ops::_Iter_equals_iter<std::reverse_iterator<__gnu_cxx::__normal_iterator<char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >(std::reverse_iterator<char const*>, std::reverse_iterator<char const*>, __gnu_cxx::__ops::_Iter_equals_iter<std::reverse_iterator<__gnu_cxx::__normal_iterator<char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, std::random_access_iterator_tag) [clone .isra.0]
        mov     rax, QWORD PTR [rsp]
        jmp     .L35
```

Notice the uninlined calls.